node/src/main/scala/coop/rchain/node/effects/package.scala

```diff
@@ -72,22 +72,52 @@ package object effects {
 
   def consoleIO(consoleReader: ConsoleReader): ConsoleIO[Task] = new JLineConsoleIO(consoleReader)
 
-  def rpConnections: Task[ConnectionsCell[Task]] =
-    Cell.mvarCell[Task, Connections](Connections.empty)
+  def rpConnections[F[_]: Concurrent]: F[ConnectionsCell[F]] =
+    Cell.mvarCell[F, Connections](Connections.empty)
+
+  def rpConfState[F[_]: Monad: Sync](conf: RPConf): MonadState[F, RPConf] =
+    new AtomicMonadState[F, RPConf](AtomicAny(conf))
+
+  def rpConfAsk[F[_]: Monad: Sync](
+      implicit state: MonadState[F, RPConf]
+  ): ApplicativeAsk[F, RPConf] =
+    new DefaultApplicativeAsk[F, RPConf] {
+      val applicative: Applicative[F] = Applicative[F]
+      def ask: F[RPConf]              = state.get
+    }
+
+  def peerNodeAsk[F[_]: Monad: Sync](
+      implicit state: MonadState[F, RPConf]
+  ): ApplicativeAsk[F, PeerNode] =
+    new DefaultApplicativeAsk[F, PeerNode] {
+      val applicative: Applicative[F] = Applicative[F]
+      def ask: F[PeerNode]            = state.get.map(_.local)
+    }
+
+  def readerTApplicativeAsk[F[_]: Monad: Sync, E, B](
```

Comment:
`Add ReaderT constructors for MonadState and ApplicativeAsk`

Those should be redundant, I'm pretty sure cats.effect/mtl have the relevant derivations.
I'm fine if we improve that in a separate PR.

Reply:
I'm not convinced about that. mapK does not exist in mtl because of reasons. vide FunctorK comments.
I'm fine in exploring further.